src/Backend/Modules/MediaGalleries/Domain/MediaGallery/MediaGallery.php

```diff
@@ -0,0 +1,403 @@
+ * @ORM\HasLifecycleCallbacks
+ */
+class MediaGallery
+{
+    /**
+     * @var string
+     *
+     * @ORM\Id()
+     * @ORM\GeneratedValue(strategy="UUID")
+     * @ORM\Column(type="guid")
+     */
+    private $id;
+
+    /**
+     * @var integer
+     *
+     * @ORM\Column(type="integer")
+     */
+    protected $userId;
+
+    /**
+     * @var integer
+     *
+     * ToDo: when Fork CMS has ModuleExtra entity in core, use its Entity as type
+     *
+     * @ORM\Column(type="integer")
+     */
+    protected $moduleExtraId;
+
+    /**
+     * @var string
+     *
+     * @ORM\Column(type="string")
+     */
+    protected $action;
+
+    /**
+     * @var string
+     *
+     * @ORM\Column(type="string")
+     */
+    protected $title;
+
+    /**
+     * @var string
+     *
+     * @ORM\Column(type="string", nullable=true)
+     */
+    protected $text;
+
+    /**
+     * @var \DateTime
+     *
+     * @ORM\Column(type="datetime")
+     */
+    protected $createdOn;
+
+    /**
+     * @var \DateTime
+     *
+     * @ORM\Column(type="datetime")
+     */
+    protected $editedOn;
+
+    /**
+     * @var \DateTime
+     *
+     * @ORM\Column(type="datetime")
+     */
+    protected $publishOn;
+
+    /**
+     * @var Status
+     *
+     * @ORM\Column(type="media_gallery_status")
+     */
+    protected $status;
+
+    /**
+     * @var MediaGroup
+     *
+     * @ORM\OneToOne(
+     *      targetEntity="Backend\Modules\MediaLibrary\Domain\MediaGroup\MediaGroup",
+     *      cascade="persist",
+     *      orphanRemoval=true
+     * )
+     * @ORM\JoinColumn(
+     *      name="mediaGroupId",
+     *      referencedColumnName="id",
+     *      onDelete="cascade"
+     * )
+     */
+    protected $mediaGroup;
+
+    /**
+     * MediaGallery constructor.
+     *
+     * @param string $title
+     * @param string $action
+     * @param int $userId
+     * @param \DateTime $publishOn
+     * @param MediaGroup $mediaGroup
+     * @param Status $status
+     * @param string|null $text
+     */
+    private function __construct(
+        string $title,
+        string $action,
+        int $userId,
+        \DateTime $publishOn,
+        MediaGroup $mediaGroup,
+        Status $status,
+        string $text = null
+    ) {
+        $this->userId = $userId;
+        $this->action = $action;
+        $this->title = $title;
+        $this->publishOn = $publishOn;
+        $this->mediaGroup = $mediaGroup;
+        $this->status = $status;
+        $this->text = $text;
+    }
+
+    /**
+     * To array
+     *
+     * @return array
+     */
+    public function __toArray(): array
+    {
+        return [
+            'id' => $this->id,
+            'userId' => $this->userId,
+            'moduleExtraId' => $this->moduleExtraId,
+            'action' => $this->action,
+            'title' => $this->title,
+            'text' => $this->text,
+            'createdOn' => $this->createdOn->getTimestamp(),
```

Comment:
why do you pass the timestamp?
wouldn't it make more sence to pass the DateTime so that if you use this in the template you can just use the format function if you want etc?